File: factory/c-sharp/NotifyFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Factory
{
    class NotifyFactory
    {
        public INotify CreateNotify(string notifyType)
        {
            if (notifyType == "SMS")
            {
                // Buradaki nesne oluşturma süreçleri bize kalmıştır.
                // Kullanıcıdan soyutlanmıştır.
                // Örnek olarak burası Singleton olarak da ayarlanabilirdi.
                return new SmsNotify();
            }
            else if (notifyType == "MAIL")
            {
                // Buradaki nesne oluşturma süreçleri bize kalmıştır.
                // Kullanıcıdan soyutlanmıştır.
                // Örnek olarak burası Singleton olarak da ayarlanabilirdi.
                return new MailNotify();
            }
            return null;
        }
    }
}

File: factory/golang/notifyFactory.go
package main

import "fmt"

func newSmsNotify(n Notify) iNotify {
	return &SmsNotify{Notify: n}
}

func newMailNotify(n Notify) iNotify {
	return &MailNotify{Notify: n}
}

func CreateNotify(notifyType string, n Notify) (iNotify, error) {
	if notifyType == "SMS" {
		return newSmsNotify(n), nil
	} else if notifyType == "MAIL" {
		return newMailNotify(n), nil
	}
	return nil, fmt.Errorf("wrong type")
}
